docs/source/_templates/sdk_TP_tutorial.rst

```diff
@@ -0,0 +1,378 @@
+            if (addresses.length === 0) {
+              throw new InternalError('State Error!')
+            }
+          })
+      })
+    }
+
+{% else %}
+{# Python code is the default #}
+
+.. code-block:: python
+
+    def apply(self, transaction, state_store):
+        signer, game_name, action, space = \
+            self._unpack_transaction(transaction)
+
+        board, state, player1, player2 = \
+            self._get_state_data(game_name, state_store)
+
+        updated_game_data = self._play_xo(
+            board, state,
+            player1, player2,
+            signer, action, space
+        )
+
+        self._store_game_data(game_name, updated_game_data, state_store)
+
+{% endif %}
+
+Note that the third step is the only one that actually concerns tic-tac-toe;
+the other three steps all concern the coordination of data.
+
+Data
+====
+
+So how do we get data out of the transaction? The transaction consists of a
+header and a payload. The header contains the "signer", which is used to
+identify the current player. The payload will contain an encoding of the game
+name, the action ('create' a game, 'take' a space), and the space (which will
+be an empty string if the action isn't 'take'). So our ``_unpack_transaction``
+function will look like this:
+
+{% if language == 'JavaScript' %}
+
+.. TODO::
+    
+        Example code to be provided by JavaScript expert
+
+
+{% else %}
+
+.. code-block:: python
+
+    def _unpack_transaction(self, transaction):
+        header = TransactionHeader()
+        header.ParseFromString(transaction.header)
+        signer = header.signer
+
+        try:
+            game_name, action, space = self._decode_data(transaction.payload)
+        except:
+            raise InvalidTransaction("Invalid payload serialization")
+
+        return signer, game_name, action, space
+
+{% endif %}
+
+
+Before we say how exactly the transaction payload will be decoded, let's look
+at ``_get_state_data``. Now, as far as the handler is concerned, it doesn't
+matter how the game data is stored. The only thing that matters is that given a
+game name, the state store is able to give back the correct game data. (In our
+full XO implementation, the game data is stored in a Merkle-radix tree.)
+
+
+{% if language == 'JavaScript' %}
+
+.. TODO:: 
+
+    Example code to be provided by JavaScript expert, along with 
+    rewrite suggestion for surrounding text.
+
+{% else %}
+
+.. code-block:: python
+
+    def _get_state_data(self, game_name, state_store):
+        game_address = self._make_game_address(game_name)
+
+        state_entries = state_store.get([game_address])
+
+        try:
+            return self._decode_data(state_entries[0].data)
+        except IndexError:
+            return None, None, None, None
+        except:
+            raise InternalError("Failed to deserialize game data.")
+
+
+{% endif %}
+
+It doesn't matter what exactly the game address is. By convention, we'll store
```

Comment:
I wouldn't say this. Addresses are soon going to be enforced to be length 70 strings of hexadecimal characters. Also, exactly where to store data is an important consideration in constructing a transaction family. It would be possible to construct a "correct" implementation of most/any transaction family by storing all information about anything that the family will ever get or set in one address, and then all txns would have to be serial no matter what.